src/main/scala/is/hail/rvd/OrderedRVDPartitioner.scala

```diff
@@ -109,6 +109,29 @@ class OrderedRVDPartitioner(
       })
     copy(numPartitions = newPartEnd.length, rangeBounds = newRangeBounds)
   }
+
+  @transient
+  @volatile var partitionerBc: Broadcast[OrderedRVDPartitioner] = _
+
+  def broadcast(sc: SparkContext): Broadcast[OrderedRVDPartitioner] = {
+    if (partitionerBc == null) {
+      synchronized {
```

Comment:
I think the `this.synchronized` pattern we use elsewhere is a good reminder that synchronization occurs on the object. Up to you, everything else looks great (clever!)

Reply:
Should we put `this` on every function we call as a reminder that it occurs on the object?  I'm not quite sure why this shows up elsewhere.  I'm not inclined to change it unless you want to push.  I will leave a note for myself to clean up the other spots.